vms/platformvm/block/executor/acceptor.go

```diff
@@ -172,6 +172,16 @@ func (a *acceptor) optionBlock(b, parent block.Block, blockType string) error {
 		return err
 	}
 
+	parentState, ok := a.blkIDToState[parentID]
+	if !ok {
+		return fmt.Errorf("%w %s", errMissingBlockState, parentID)
+	}
+	if parentState.onDecisionState != nil {
+		if err := parentState.onDecisionState.Apply(a.state); err != nil {
```

Comment:
To confirm, just calling the `onAcceptState` doesn't commit all diffs under it (for example the `wrapped decision state`)?

Reply:
That's correct

Reply:
We could apply the `onAcceptState` to `onDecisionState` and then apply `onDecisionState` to `a.state` if you felt that was cleaner... But this is the (slightly) more efficient way to do the same thing.

Reply:
I think what we have here is fine. Makes sense (should be the same thing).